viewer/packages/data-providers/src/image-360-data-providers/Cdf360ImageEventProvider.ts

```diff
@@ -0,0 +1,118 @@
+/*!
+ * Copyright 2022 Cognite AS
+ */
+
+import * as THREE from 'three';
+import groupBy from 'lodash/groupBy';
+import orderBy from 'lodash/orderBy';
+import zipWith from 'lodash/zipWith';
+
+import { CogniteClient, FileInfo } from '@cognite/sdk';
+import { Image360Descriptor, Image360Face } from '../types';
+import { Matrix4 } from 'three';
+import { Image360Provider } from '../Image360Provider';
+import assert from 'assert';
+
+type Event360Metadata = Event360Filter & Event360TransformationData;
+
+type Event360TransformationData = {
+  rotation_angle: string;
+  rotation_axis: string;
+  translation: string;
+};
+
+type Event360Filter = {
+  site_id: string;
+  site_name: string;
+  station_id: string;
+  station_name: string;
+};
+
+export class Cdf360ImageEventProvider implements Image360Provider<{ [key: string]: string }> {
+  private readonly _client: CogniteClient;
+  constructor(client: CogniteClient) {
+    this._client = client;
+  }
+
+  public async get360ImageDescriptors(metadataFilter: { [key: string]: string }): Promise<Image360Descriptor[]> {
+    const image360Events = await this._client.events.list({ filter: { metadata: metadataFilter } });
+    return image360Events.items
+      .map(image360Event => image360Event.metadata as Event360Metadata)
+      .map(eventMetadata => this.parseEventMetadata(eventMetadata));
+  }
+
+  public async get360ImageFiles(image360Descriptor: Image360Descriptor): Promise<Image360Face[]> {
+    const { collectionId, id } = image360Descriptor;
+    const fileInfos = await this.getFileInfos(collectionId, id);
+    const fileIds = fileInfos.map(fileInfo => {
+      return { id: fileInfo.id };
+    });
+    const fileBuffers = await this.getFileBuffers(fileIds);
+
+    const faces = zipWith(fileInfos, fileBuffers, (fileInfo, fileBuffer) => {
+      return {
+        face: fileInfo.metadata!.face,
+        data: fileBuffer
+      } as Image360Face;
+    });
+
+    return faces;
+  }
+
+  private async getFileInfos(siteId: string, stationId: string): Promise<FileInfo[]> {
+    const fileInfos = await this._client.files.list({
+      filter: {
+        uploaded: true,
+        metadata: {
+          site_id: siteId,
+          station_id: stationId,
+          image_type: 'cubemap'
+        }
+      }
+    });
+
+    assert(fileInfos.items.length > 0);
+
+    if (fileInfos.items[0].metadata?.timestamp !== undefined) {
```

Comment:
What is happening here? Multiple possible return formats we need to handle?

Reply:
The datamodel can contain a timestamp, refactored it out for easier readability.